### PR TITLE
feat(panel): Inject mdPanelRef when instantiating the panel controller.

### LINKED ISSUE #7177: mdPanel: Inject mdPanelRef when instantiating the panel controller.
Figure out how to do this:

1. mdPanelRef doesn't exist until the panel is created.
2. mdPanelRef needs to be injected during creation.

### PR DESCRIPTION
**Pull request by KarenParker**
Fixes #7177.

@ErinCoughlan @gmoothart @DerekLouie @jelbourn Please review.

### CODE REVIEW

**Comment by jelbourn** [comment]
LGTM

File: src/components/panel/panel.spec.js [hunk 1]
```diff
@@ -597,12 +598,13 @@ describe('$mdPanel', function() {
 
     it('should allow locals to be injected in the controller', function() {
```
Comment (ErinCoughlan) [inline src/components/panel/panel.spec.js]: optional: Add a test that a panel can close itself. That's the main benefit (I think) of injecting MdPanelRef.

**Comment by ErinCoughlan** [comment]
LGTM

File: src/components/panel/demoBasicUsage/script.js [hunk 2]
```diff
@@ -32,10 +29,12 @@ BasicDemoCtrl.prototype.showDialog = function() {
 };
 
 
-BasicDemoCtrl.prototype.closeDialog = function() {
-  this._panelRef && this._panelRef.close();
-};
+// Necessary to pass locals to the dialog template.
+function PanelDialogCtrl(mdPanelRef) {
```
Comment (ThomasBurleson) [inline src/components/panel/demoBasicUsage/script.js]: I strongly suggest a different name for `mdPanelRef`.

*  This is a reference to some instance ?
*  What type of instance does this reference ?
*  `Ref` seems very cludgy

**Comment by jelbourn** [comment]
@ThomasBurleson `Ref` is the term Angular 2 uses for this sort of thing (`ComponentRef`, `ApplicationRef`, `ElementRef`, `TemplateRef`, etc.). The naming simply says "This is a reference to some existing panel." I think it does a good job of conveying what you're injecting.

**Pull request closed without merging**